### DISCUSSION TITLE
How do I properly add a className to a <Link> tag?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#44755] [Help] [answered]
This doesn't work:

```
<div>
Click here to <Link className ="link" href="/contact-us"> contact us</Link>
</div> 
```

But if I replace the `<Link>` tag for an `<a> `tag it works fine, but I just have to add the entire https://www  information..

**Comment by icyJoseph** [comment, marked as answer]
What next.js version are you using? 

Starting with Next.js 12, changes were introduced to the Link component, but the changes were opt-in.

Starting with Next.js 13, the changes  were set by default, but you could opt-out.

Let's call the default behavior up to 12, legacy, and from 13+, new behavior.

This is important because the legacy link behavior did not allow setting the class name on the link tag, rather, you had to provide an anchor tag as children and pass a class name to it:

```jsx
// legacy
<Link href="foo">
   <a className="my-className">Hi</a>
</Link>
```

The new link behavior, among other things, allows you to pass the classname directly on the link:

```jsx
<Link href="foo" className="my-ClassName">Hi</Link>
```

**Reply by Elindo586** [reply]
Silly question... how do I check which version of Next am I running???

And how can I update it?

And how do I go back if anything gets messed up? 

Ideally I would like to keep everything on the latest versions, that would include dependencies as well which I check from time to time.

**Reply by icyJoseph** [reply]
Look at your `package.json`.

Alternatively, `npm explain next` or `yarn why next` would tell you exactly which next.js version you got.

To update to latest `npm i next@latest` or `yarn add next@latest` 

To rollback, `npm i next@12.1.1` for example, that would set it back tp 12.1.1.

Not to mention that you can checkout git changes, even revert commits, unstage changes etc.

I do think you had next 12 so you need to have a child anchor tag that takes in the class name.

**Reply by Elindo586** [reply]
I think this is the one...

  "dependencies": {
    "bootstrap": "^5.2.3",
    "gsap": "^3.11.0",
    "next": "^12.1.6",
    "nodemailer": "^6.7.5",
    "react": "18.1.0",
    "react-bootstrap": "^2.7.0",
    "react-dom": "18.1.0",
    "sass": "^1.53.0",
    "sharp": "^0.30.7"
  },

**Reply by Elindo586** [reply]
I tried to update it and it gave me this:

```
PS C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb> npm i next@latest
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: nextprojectb@0.1.0
npm ERR! Found: react@18.1.0
npm ERR! node_modules/react
npm ERR!   react@"18.1.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.2.0" from next@13.1.1
npm ERR! node_modules/next
npm ERR!   next@"13.1.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\Edgar\AppData\Local\npm-cache\_logs\2023-01-11T00_38_41_497Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Edgar\AppData\Local\npm-cache\_logs\2023-01-11T00_38_41_497Z-debug-0.log
PS C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb> 
```

**Reply by GabenGar** [reply]
You have to update `react` also.
